mmdet/evaluation/metrics/crowdhuman_metric.py

```diff
@@ -0,0 +1,822 @@
+            matched_indices = np.array([j for (_, j) in matches])
+            unmatched_indices = list(
+                set(np.arange(gt_boxes[gt_tag].shape[0])) -
+                set(matched_indices))
+            num_ignore_gt = self.get_ignores(
+                gt_boxes[gt_tag][unmatched_indices], gt_boxes[~gt_tag, :4])
+            # compute results
+            eps = 1e-6
+            k = len(matches)
+            m = gt_tag.sum() - num_ignore_gt
+            n = dt_boxes.shape[0] - num_ignore_dt
+            ratio = k / (m + n - k + eps)
+            recall = k / (m + eps)
+            cover = k / (n + eps)
+            noise = 1 - cover
+            result_dict = dict(
+                ratio=ratio,
+                recall=recall,
+                cover=cover,
+                noise=noise,
+                k=k,
+                m=m,
+                n=n)
+            result_queue.put_nowait(result_dict)
+
+    @staticmethod
+    def gather(results):
+        """Integrate test results."""
+        assert len(results)
+        img_num = 0
+        for result in results:
+            if result['n'] != 0 or result['m'] != 0:
+                img_num += 1
+        mean_ratio = np.sum([rb['ratio'] for rb in results]) / img_num
+        valids = np.sum([rb['k'] for rb in results])
+        total = np.sum([rb['n'] for rb in results])
+        gtn = np.sum([rb['m'] for rb in results])
+        line = 'mean_ratio:{:.4f}, valids:{}, total:{}, gtn:{}'\
+            .format(mean_ratio, valids, total, gtn)
+        return line, mean_ratio
+
+    def compute_ji_matching(self, dt_boxes, gt_boxes):
+        """Match the annotation box for each detection box.
+
+        Args:
+            dt_boxes(ndarray): Detection boxes.
+            gt_boxes(ndarray): Ground_truth boxes.
+
+        Returns:
+            matches_(list[tuple[int, int]]): Match result.
+        """
+        assert dt_boxes.shape[-1] > 3 and gt_boxes.shape[-1] > 3
+        if dt_boxes.shape[0] < 1 or gt_boxes.shape[0] < 1:
+            return list()
+
+        ious = bbox_overlaps(dt_boxes, gt_boxes, mode='iou')
+        input_ = copy.deepcopy(ious)
+        input_[input_ < self.iou_thres] = 0
+        match_scipy = maximum_bipartite_matching(
+            csr_matrix(input_), perm_type='column')
+        matches_ = []
+        for i in range(len(match_scipy)):
+            if match_scipy[i] != -1:
+                matches_.append((i, int(match_scipy[i])))
+        return matches_
+
+    def get_ignores(self, dt_boxes, gt_boxes):
+        """Get the number of ignore bboxes."""
+        if gt_boxes.size:
+            ioas = bbox_overlaps(dt_boxes, gt_boxes, mode='iof')
+            ioas = np.max(ioas, axis=1)
+            rows = np.where(ioas > self.iou_thres)[0]
+            return len(rows)
+        else:
+            return 0
+
+
+class Image(object):
+    """Data structure for evaluation of CrowdHuman. Please refer to
+    https://github.com/Purkialo/CrowdDet for more details.
```

Comment:
The reference here is not enough. Seems that the code is modified from https://github.com/Purkialo/CrowdDet/blob/master/lib/evaluate/APMRToolkits/image.py. If so, we should add a note here:

```
"""Data structure for evaluation of CrowdHuman. 

Note:
    This implementation is modified from https://github.com/Purkialo/CrowdDet/blob/master/lib/evaluate/APMRToolkits/image.py

Args: